### PR TITLE
#65 - purge mode

### LINKED ISSUE #65: Create separate config for cleaning up fresh Laravel installation
A fresh Laravel installation has a lot of useless stuff like stupid comments, no return types, useless methods and so on. It would be nice if we could create a separate config with prefabricated set of rules to clean this up.

This could be used in normal `codestyle.php` file like that:
```php
<?php

declare(strict_types=1);

use Blumilk\Codestyle\FreshLaravelConfig;
use Blumilk\Codestyle\Configuration\Defaults\Paths;

$config = new FreshLaravelConfig();
return $config->config();
```

or maybe:
```php
<?php

declare(strict_types=1);

use Blumilk\Codestyle\Config;
use Blumilk\Codestyle\Configuration\Defaults\Paths;

$config = new Config();
return $config->freshLaravelConfig();
```

Or maybe even as a script in `./vendor/bin/codestyle --laravel`.

### PR DESCRIPTION
**Pull request by krzysztofrewak**
For new Laravel instances sometimes I would like yo get rid of all comments, especially in configuration files. This should allow it by entering "purge mode" in `codestyle.php` for one run:
```php
return $config->purgeMode()->config();
```

With purge mode this:
```php
return [
    /*
    |--------------------------------------------------------------------------
    | Default Database Connection Name
    |--------------------------------------------------------------------------
    |
    | Here you may specify which of the database connections below you wish
    | to use as your default connection for all database work. Of course
    | you may use many connections at once using the Database library.
    |
    */

    "default" => env("DB_CONNECTION", "mysql"),

    /*
    |--------------------------------------------------------------------------
    | Database Connections
    |--------------------------------------------------------------------------
    |
    | Here are each of the database connections setup for your application.
    | Of course, examples of configuring each database platform that is
    | supported by Laravel is shown below to make development simple.
    |
    |
    | All database work in Laravel is done through the PHP PDO facilities
    | so make sure you have the driver for your particular database of
    | choice installed on your machine before you begin development.
    |
    */

    "connections" => [
        "sqlite" => [
            "driver" => "sqlite",
            "url" => env("DATABASE_URL"),
            "database" => env("DB_DATABASE", database_path("database.sqlite")),
            "prefix" => "",
            "foreign_key_constraints" => env("DB_FOREIGN_KEYS", true),
        ],
```

will be refactored into this:
```php
return [
    "default" => env("DB_CONNECTION", "mysql"),

    "connections" => [
        "sqlite" => [
            "driver" => "sqlite",
            "url" => env("DATABASE_URL"),
            "database" => env("DB_DATABASE", database_path("database.sqlite")),
            "prefix" => "",
            "foreign_key_constraints" => env("DB_FOREIGN_KEYS", true),
        ],
```

This should close #65.

### CODE REVIEW

File: tests/codestyle/fixtures/noComments/expected.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Support\Str;
+
+return [
+    "default" => env("DB_CONNECTION", "mysql"),
+
```
[the remark this answers is not in the document]
Comment (krzysztofrewak) [reply tests/codestyle/fixtures/noComments/expected.php:9]: I don't get it why it's working like this. According to this (https://mlocati.github.io/php-cs-fixer-configurator/#version:3.8|fixer:no_extra_blank_lines example 6) these empty lines should be removed.

[the remark this answers is not in the document]
Comment (krzysztofrewak) [reply tests/codestyle/fixtures/noComments/expected.php:9]: I created a task for that: https://github.com/blumilksoftware/codestyle/issues/87

File: src/Fixers/NoCommentFixer.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+use PhpCsFixer\Fixer\FixerInterface;
+use PhpCsFixer\Fixer\Whitespace\NoExtraBlankLinesFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixerCustomFixers\TokenRemover;
+use SplFileInfo;
+
+final class NoCommentFixer implements FixerInterface
```
Comment (mtracz) [inline src/Fixers/NoCommentFixer.php:16]: Will it remove all comments in the code? E.g.
```php
// some helpful info about this var
protected int $xyz = 1;
```

File: src/Configuration/Defaults/CommonRules.php [hunk 3]
```diff
@@ -113,7 +112,6 @@ class CommonRules extends Rules
         ArrayIndentationFixer::class => true,
         TrimArraySpacesFixer::class => true,
         WhitespaceAfterCommaInArrayFixer::class => true,
-        NoTrailingCommaInSinglelineArrayFixer::class => true,
```
[the remark this answers is not in the document]
Comment (krzysztofrewak) [reply src/Configuration/Defaults/CommonRules.php:116]: It's deprecated.

```php

/**
 * @deprecated
 *
 * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
 * @author Sebastiaan Stok <s.stok@rollerscapes.net>
 */
final class NoTrailingCommaInSinglelineArrayFixer extends AbstractProxyFixer implements DeprecatedFixerInterface
{
```

File: src/Fixers/NoCommentFixer.php (hunk 2 above)
Comment (krzysztofrewak) [reply src/Fixers/NoCommentFixer.php:16]: Yes. That's why it's intended only for one run for basic Laravel installation. It should not be used during development later and that's why it's not in `CommonRules`.

**Comment by kamilpiech97** [comment]
conflicts @krzysztofrewak

**Comment by krzysztofrewak** [comment]
> conflicts @krzysztofrewak

Resolved.

**Review by kamilpiech97 — APPROVED** [review]

**Review by mtracz — APPROVED** [review]

**Pull request merged**